src/utils/convert.js

```diff
@@ -212,18 +212,85 @@ const renderAuth = (definition) => {
   }
 };
 
+// Check if scripting has a given method for a step type, key, and position (pre, post, full)
+const hasScriptingMethod = (js, type, key, position) => {
+  if (!js) {
+    return false;
+  }
+
+  let methodSuffix = '';
+
+  if (type === 'trigger') {
+    if (position === 'pre') {
+      methodSuffix = '_pre_poll';
+    } else if (position === 'post') {
+      methodSuffix = '_post_poll';
+    } else {
+      methodSuffix = '_poll';
+    }
+  } else if (type === 'create') {
+    if (position === 'pre') {
+      methodSuffix = '_pre_write';
+    } else if (position === 'post') {
+      methodSuffix = '_post_write';
+    } else {
+      methodSuffix = '_write';
+    }
+  } else if (type === 'search') {
+    if (position === 'pre') {
+      methodSuffix = '_pre_search';
+    } else if (position === 'post') {
+      methodSuffix = '_post_search';
+    } else {
+      methodSuffix = '_search';
+    }
+  }
+
+  const methodName = `${key}${methodSuffix}`;
+
+  return (js.indexOf(methodName) !== -1);
```

Comment:
Maybe we want to strip out the comments from `js` first, but we can do it in a separate PR.

Reply:
Yes, great idea!